crypto/cms/cms_ec.c

```diff
@@ -50,16 +54,17 @@ static EVP_PKEY *pkey_type2param(int ptype, const void *pval,
         }
         if (EVP_PKEY_paramgen(pctx, &pkey) <= 0)
             goto err;
-    } else {
-        ERR_raise(ERR_LIB_CMS, CMS_R_DECODE_ERROR);
-        goto err;
+        EVP_PKEY_CTX_free(pctx);
+        return pkey;
     }
 
+    ERR_raise(ERR_LIB_CMS, CMS_R_DECODE_ERROR);
     return pkey;
```

Comment:
A `return NULL` would be more clear here, but nevertheless okay.

Reply:
I'll change it.